week-3/Homework/mandatory/4-eligible-students.js

```diff
@@ -1,25 +1,147 @@
+    [
+      ["Hunor", 10],
+      ["Magor", 1]
+    ]
+    See that each student's information is stored separately as an embedded array.
+    This is also called as 2 dimensional array when you can find array inside of array.
+
   - Returns an array containing only the names of the who have attended AT LEAST 8 classes
+
+  
 */
 
-function eligibleStudents() {
+// Remember how to access to embedded arrays
+
+let twoDimensionalArray = [
+  ["cat", "dog"],
+  ["giraffe", "lion", "elephant"]
+];
+
+// Examples accessing to array element directly
+// Example 1
+let pets = twoDimensionalArray[0]; // this reads the first embedded array 
+console.log(pets[1]); // this prints "cat" to the console
+
+// Example 2
+// You can also read straight the element of an embedded array:
+console.log(twoDimensionalArray[1][2]);
+// This reads the second embedded array and then reads its last element and finally prints "elephant" to the console
+
+// Examples accessing to array element through array methods
+// Example 1
+let moreThanTwoArrays = twoDimensionalArray.filter(embeddedArray => embeddedArray.length > 2);
+console.log(moreThanTwoArrays);
+// This filter only keeps embededd arrays that have more than 3 elements
+
+// Example 2
+let arrayLengths = twoDimensionalArray.map(embeddedArray => embeddedArray.length);
+console.log(arrayLengths);
+// This gives back how many elements of each array have
+
+/* + + + + + + + */
+/* Now try the exercise */
+//refrence -https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Array_destructuring
+function getData(people){
+  let studentAttednded8Class=people.filter(function(personalInfo){
+    return (personalInfo[1]>=8);
+  
+ });
+      return studentAttednded8Class;
 
 }
+function getEligibleStudents(people) {
+      let studentData=getData(people);
+     var names=studentData.map(function(info){
+     return info[0]
+});
+    
+  return names ;
+}
+
+
+console.log(getEligibleStudents([
+  ["Ahmed", 8],
+  ["Clement", 10],
+  ["Elamin", 6],
+  ["Adam", 7],
+  ["Tayoa", 11],
+  ["Nina", 10],
+  ["Bob", 9],
+  ["Lee", 1]
+])
+);
+
+/*
+ 
+ Later leaders of Alpha planet decided to change the rule as below:
+
+ Only students whose name starts with the same letter of the name of the planet are eligible to join.
+
+ Implement the body of function called "getEligibleStudents2" which:
+  - Accepts an array in the same structure as before.
+  - Returns an array containing only the names that satisfies the new rule.
+  
+  Note:
+  Unfortunately, administrators messed up the letter casing of names, sometimes it starts with small letter.
+
+  Hint: To complete the function, search how to change text to lower or upper case by using string method.
+*/
+
+function getEligibleStudents2(eligibleStudentNameWithA) {
+  let studentName=getEligibleStudents(eligibleStudentNameWithA);
+  let correctNames=studentName.filter(function(allNames){
+    return allNames.startsWith('A' || 'a');
```

Comment:
This won't work as you expect because you need to use two calls to `startsWith()` in order to get this working:
```suggestion
    return allNames.startsWith('A') || allNames.startsWith('a');
```
Also think about using a different variable name other than `allNames` - this is plural at the moment but it would be better to use something **singular**